test/fixtures/pretty/child-with-updated-chindings.js

```diff
@@ -0,0 +1,8 @@
+global.process = { __proto__: process, pid: 123456 }
+Date.now = function () { return 1459875739796 }
+require('os').hostname = function () { return 'abcdefghijklmnopqr' }
+var pino = require(require.resolve('./../../../'))
+var log = pino({ prettyPrint: true }).child({ foo: 123 })
+log.info('before')
+log.setBindings({ foo: 456, bar: 789 })
```

Comment:
Are we really exposing this as a public API? Can we exercise the code path without using this function?

Reply:
Which function are you referring to, `.setBindings()`? The point of this PR is to make that method work correctly when a prettifier is in use.

Reply:
setBindings is already exposed.

Reply:
I don't see it in our docs -- https://getpino.io/#/docs/api

Reply:
Maybe it should be documented, anyway this PR does not add it.